File: Arrays/Medium/Kth Maximum and Minimum Element/Kth_max_and_min_element.java
import java.util.Arrays;

public class Kth_max_and_min_element {
    /*
     * Function to find kth minimum Element of the Array. Tc:- O(n log(n)), SC:-
     * O(1).
     */
    static int findKthMinimum(int arr[], int k) {
        Arrays.sort(arr);
        return arr[k - 1];
    }

    /*
     * Function to find kth maximum Element of the Array. Tc:- O(n log(n)), SC:-
     * O(1).
     */
    static int findKthMaximum(int arr[], int k) {
        Arrays.sort(arr);
        return arr[arr.length - k];
    }

    public static void main(String[] args) {
        int[] arr = { 10, 20, 3, 2, 6, 59, 87, 1 };
        int k = 2;

        System.out.println("Kth Maximum Element is " + findKthMaximum(arr, k));
        System.out.println("Kth Minimum Element is " + findKthMinimum(arr, k));
    }
}

File: 1. Arrays/Medium/Kth Maximum and Minimum Element/Kth_max_and_min_element.cpp
#include <bits/stdc++.h>
using namespace std;

/* Function to find kth minimum Element of the Array. Tc:- O(n log(n)), SC:- O(1).  */
int findKthMinimum(vector<int> &arr, int k)
{
    sort(arr.begin(), arr.end());
    return arr[k - 1];
}

/* Function to find kth maximum Element of the Array. Tc:- O(n log(n)), SC:- O(1).  */
int findKthMaximum(vector<int> &arr, int k)
{
    sort(arr.begin(), arr.end());
    return arr[arr.size() - k];
}

int main()
{
    vector<int> arr = {10, 20, 3, 2, 6, 59, 87, 1};
    int k = 2;

    cout << "Kth Maximum Element:- " << findKthMaximum(arr, k) << endl;
    cout << "Kth Minimum Element:- " << findKthMinimum(arr, k) << endl;
}